src/modules/dashboard/components/CoinMachine/RemainingDisplayWidget/RemainingDisplayWidget.tsx

```diff
@@ -105,17 +125,76 @@ const RemainingDisplayWidget = ({
     if (displayType === DataDisplayType.Time && splitTime) {
       return <TimerValue splitTime={splitTime} />;
     }
-    if (value && displayType !== DataDisplayType.Time) {
-      return value;
+    if (periodTokens && displayType === DataDisplayType.Tokens) {
+      const { soldPeriodTokens, decimals, maxPeriodTokens } = periodTokens;
+
+      if (soldPeriodTokens.gte(maxPeriodTokens)) {
+        return <FormattedMessage {...MSG.soldOut} />;
+      }
+
+      return `${getFormattedTokenValue(
+        maxPeriodTokens.sub(soldPeriodTokens),
+        decimals,
+      )}/${getFormattedTokenValue(maxPeriodTokens, decimals)}`;
     }
 
     return <FormattedMessage {...widgetText.placeholder} />;
-  }, [displayType, splitTime, value, widgetText]);
+  }, [displayType, splitTime, widgetText, periodTokens]);
+
+  const priceStatus = useMemo(() => {
+    if (!periodTokens || displayType === DataDisplayType.Time) {
+      return undefined;
+    }
+
+    const {
+      soldPeriodTokens,
+      maxPeriodTokens,
+      targetPeriodTokens,
+    } = periodTokens;
+
+    if (soldPeriodTokens.gte(maxPeriodTokens)) {
+      return TokenPriceStatuses.PRICE_SOLD_OUT;
+    }
+
+    if (soldPeriodTokens.eq(targetPeriodTokens)) {
+      return TokenPriceStatuses.PRICE_NO_CHANGES;
+    }
+
+    if (soldPeriodTokens.lt(targetPeriodTokens)) {
+      return TokenPriceStatuses.PRICE_DOWN;
+    }
+
+    if (soldPeriodTokens.gt(targetPeriodTokens)) {
+      return TokenPriceStatuses.PRICE_UP;
+    }
+
+    return undefined;
+  }, [displayType, periodTokens]);
+
   const showValueWarning =
+    displayType === DataDisplayType.Time &&
     appearance.theme !== 'danger' &&
     typeof value === 'number' &&
     periodLength &&
-    ((value / 1000) * 100) / periodLength <= 10;
+    (timeLeft * 100) / periodLength <= 10;
+
+  const isValueWarning = useMemo(() => {
+    if (periodTokens?.soldPeriodTokens.gte(periodTokens?.maxPeriodTokens)) {
+      return true;
+    }
+
+    return false;
+  }, [periodTokens]);
+
+  useEffect(() => {
+    if (timeLeft === 0 && colonyAddress !== undefined) {
+      dispatch({
+        type: ActionTypes.COIN_MACHINE_PERIOD_UPDATE,
+        payload: { colonyAddress },
+      });
```

Comment:
I think that we need it as the Tokens & Time parts should be separated. They just create messy code when they are together. That will be part of my refactor that I was talking about on discord.

Reply:
so if you will have addiitonal query, then you will use refetch from additional query, not additional saga for updating

Reply:
You can't do that, because the query needed won't be in the component and I don't think it's good to pass it via a prop to components (you said in your own comment to the other PR that it's good to avoid prop drilling). It's the same principle as in motions: time changes in one component & the query needs to be updated in a different component. You don't have this query now in that component anyway, it's in the coinMachine.

Reply:
Ok so maybe im not sure what you plan to do, but i will see in the PR itself. For now please use refetch provided by use query hook

Reply:
I am stating again that the query is not even in the component. It's in CoinMachine. Do you want me to pass it as a prop? I disagree with this approach.

Reply:
I actually agree with @chinins 's approach here. While I'm not a fan of calling sagas from components, in this case make sense.

As for just having this one saga just to update the time it makes it much better, as the code won't need to update values it won't use, so the update _should_ happen faster.

On the other hand, if we put more and more queries in this saga _(down the line)_, we might need to think about combining it with others, but for now I think it's OK if it stays like this.

However, I would make the time comparison something like:
```js
if (timeLeft <= 0 && colonyAddress) { ... }
```
This is just to guard against `timeLeft` underflowing below `0` _(something like `-1` or `-2`)_ in which case the logic check will break

Reply:
You can use this query in component itself, instead of passing props to the child. It will fetch from cached data so it wont cost anything

Reply:
That won't work as I need this query in `CoinMachine` for another CoinMachine child component.

I also think that Raul agreed with my approach so could we move on with this PR if you don't have any other objections?

Reply:
i mean you can use this query whenever you want, so what is the problem? We dont need additional resolvers and query in our codebase, we should use existing approaches.

Reply:
Yeah, but then I need it to be updated when the timer runs out and I will need this anyway. Or I can use it in CoinMachine once and then pass 1 value as prop to 2-3 components that need that. And use this refetching anyway as well. In any case I need this refetching mechanism.

We are wasting time here, it's become a too long and unproductive discussion. Can we already proceed? There are PRs that wait for it to be merged.

Reply:
We dont need additional code in our codebase. https://github.com/JoinColony/colonyDapp/blob/60026cf815138dc721e7115953d96abc0c16f4f1/src/modules/dashboard/components/ActionsPage/CountDownTimer/CountDownTimer.tsx#L179 this mechanism is exactly provided for cases like yours. And you said lastly that its important to have good codebase.

Reply:
I understand the method and I agree that it's important to have a good code base but I don't think that refetch should be used here as the refetch should be triggered in a component(s) different from where the query is. Unless we pass it as a prop which I don't think is a good idea.